arwen/host/arwen.go

```diff
@@ -35,27 +35,27 @@ type vmHost struct {
 	storageContext    arwen.StorageContext
 	bigIntContext     arwen.BigIntContext
 
-	scAPIMethods *wasmer.Imports
+	scAPIMethods             *wasmer.Imports
+	protocolBuiltinFunctions vmcommon.FunctionNames
 }
 
 // NewArwenVM creates a new Arwen vmHost
 func NewArwenVM(
 	blockChainHook vmcommon.BlockchainHook,
 	cryptoHook vmcommon.CryptoHook,
-	vmType []byte,
-	blockGasLimit uint64,
-	gasSchedule map[string]map[string]uint64,
+	hostParameters *arwen.VMHostParameters,
 ) (*vmHost, error) {
 
 	host := &vmHost{
-		blockChainHook:    blockChainHook,
-		cryptoHook:        cryptoHook,
-		meteringContext:   nil,
-		runtimeContext:    nil,
-		blockchainContext: nil,
-		storageContext:    nil,
-		bigIntContext:     nil,
-		scAPIMethods:      nil,
+		blockChainHook:           blockChainHook,
+		cryptoHook:               cryptoHook,
```

Comment:
Please add checks against `nil` values for the received `blockChainHook` and `cryptoHook`, so the constructor can return an error. I forgot to add them previously.